src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Everything/PatientEverythingService.cs

```diff
@@ -177,6 +223,45 @@ public class PatientEverythingService : IPatientEverythingService
             return new SearchResult(searchResultEntries, searchResult.ContinuationToken, searchResult.SortOrder, searchResult.UnsupportedSearchParameters);
         }
 
+        private void StoreSeeAlsoLinkInformationInToken(string resourceId, EverythingOperationContinuationToken token, IReadOnlyCollection<SearchResultEntry> searchResultEntries)
+        {
+            SearchResultEntry patientResource = searchResultEntries.FirstOrDefault(s => string.Equals(s.Resource.ResourceTypeName, ResourceType.Patient.ToString(), StringComparison.Ordinal));
+
+            if (patientResource.Resource != null)
+            {
+                ResourceElement element = _resourceDeserializer.Deserialize(patientResource.Resource);
+                Patient patient = element.ToPoco<Patient>();
+
+                List<Patient.LinkComponent> links = patient.Link;
+
+                if (links != null)
+                {
+                    foreach (Patient.LinkComponent link in links)
+                    {
+                        ReferenceSearchValue referenceSearchValue = _referenceSearchValueParser.Parse(link.Other.Reference);
+
+                        // If the link does not point back to the current patient
+                        if (!string.Equals(referenceSearchValue.ResourceId, resourceId, StringComparison.Ordinal))
+                        {
+                            if (link.Type == Patient.LinkType.ReplacedBy)
+                            {
+                                throw new EverythingOperationException(string.Format(Core.Resources.EverythingOperationResourceIrrelevant, resourceId, referenceSearchValue.ResourceId), HttpStatusCode.BadRequest);
```

Comment:
Is `BadRequest` the most appropriate status code for this scenario?

Reply:
This coming from encrypted token, right?

What I'm as user suppose to do with this message?

Reply:
Maybe we can change it from BadRequest to Moved Permanently (301) and provide updated url in Location header?

Reply:
> What I'm as user suppose to do with this message?

The user would know that the patient they are currently running the $everything operation for has been replaced. They would run patient $everything on the id provided in the operation outcome

Reply:
> This coming from encrypted token, right?

I'm not sure what you mean by this exactly!

If it's helpful, here is a bit more context about this line of code: this will be reached in phase 0, after the patient has been retrieved. This method processes the links in the patient resource, and if a `replaced-by` link is found, an operation outcome is returned.

Reply:
resourceId = token.ProcessingSeeAlsoLink ? token.CurrentSeeAlsoLinkId : resourceId;

there is case where resourceId taken from token, and that token produced by us, so it's wrapped in base64, right?

Reply:
The token is decoded at the beginning of the `SearchAsync()` method:

```
EverythingOperationContinuationToken token = string.IsNullOrEmpty(continuationToken)
    ? new EverythingOperationContinuationToken(0, null)
    : EverythingOperationContinuationToken.FromString(ContinuationTokenConverter.Decode(continuationToken));
```

Reply:
so there is two id's one for Patient/id/everything and there is do for seelso patient.

Patient-id is provided by user, it's part of api call.
seealso-id is provided by us and it's encoded in continuation token.

If this link talks about patient-id, it's ok to send that message to user because user can change id to appropriate one.

But if this is seealso id, user can't do much, because it's outside of user control.

Reply:
As discussed, this code is only reached in Phase 0 of the first patient $everything call. I'll add some comments to make this more clear.

Reply:
Added comments in a8f5b52.